### ISSUE TITLE
MD032 is not triggered when list is not followed by a blank line

### ISSUE THREAD

**Issue by Vlad1m1ru5** [DavidAnson/markdownlint#428]
# MD032 is not triggered when list is not followed by a blank line

| Package        | Version   |
| :------------- | :-------- |
| `markdownlint` | `~0.23.1` |

## How to Reproduce

1. Create .md file with following content from [documentation](https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md032)

    ```md
    # List blanks

    Some text
    - Some
    - List

    1. Some
    2. List
    Some text

   ```

2. Process it with markdownlint

## Expected Behavior

```sh
4 - MD032 / blanks-around-lists Lists should be surrounded by blank lines
8 - MD032 / blanks-around-lists Lists should be surrounded by blank lines
```

## Actual Behavior

```sh
4 - MD032 / blanks-around-lists Lists should be surrounded by blank lines
```

**Comment by DavidAnson** [comment]
The output of the linter is correct, line 8 is part of list item 2.

https://dlaa.me/markdownlint/#%25m%23%20List%20blanks%0A%0ASome%20text%0A-%20Some%0A-%20List%0A%0A1.%20Some%0A2.%20List%0ASome%20text%0A

The specification calls this "lazy continuation lines" and example 291 is the most relevant.

https://spec.commonmark.org/0.30/#lazy-continuation-line

**Comment by NomarCub** [comment]
For others like me, looking for more info, this came up previously here: #211, #212.
